code/API_definitions/Traffic_Influence.yaml

```diff
@@ -135,32 +135,40 @@ info:
     "trafficInfluenceID"). All the created resources are aggregated by the
     Application (identified by "appId").\
     \
-    **appId**
+    **appId:**
     A globally unique identifier associated with the application. This
     identifier is provided during the application onboarding process.
     To influence the traffic toward a specific Application. It is the Operator
     Platform that detects the appropriate Application instance in the selected
     "region" or "zone".\
     \
-    **appInstanceId**
+    **appInstanceId:**
     A globally unique identifier generated by the Operator Platform to identify
     a specific instance of the Application on a specific zone. To influence a
     traffic toward a specific Application instance.\
     \
-    **trafficFilters**
-    The Application can expose different service on different interfaces, with
-    this parameter it is possible to enable just some of those services maybe
-    for different sets of users.\
+    **sourceTrafficFilters:**
+    The traffic can be from a specific application port in the Device.\
```

Comment:
I think we need to differentiate qualify between "application" in sourceTrafficFilters and one in destinationTrafficFilters description. While the application in sourceTrafficFilters seems to refer to  device the destinationTrafficFilters could be referring to EAS. This is to avoid any confusion w.r.t terms we have defined so far

Reply:
you are right

Reply:
we have:

source private ip and port
source public ip and port
destination ip, port and protocol

in sourceTrafficFilter we have the source private port.

we need to better specify this.

Reply:
I'm wondering @babunkj how the UPF can be effected by the "source private ip"

Reply:
> I'm wondering @babunkj how the UPF can be effected by the "source private ip"

The packet Detection Rules in the UPF can include the source device Ip address and port and the destination IP Address and port.

Reply:
We can use 'specific source port in the Device' or 'specific application client port in the Device' instead of 'specific application port in the Device'.

Reply:
@babunkj sure, we will not use APPLICATION port

Anyway, it is totally clear to me the "source port" is required, my doubt is about "source PRIVATE port". The UPF is indeed effected by the "source PUBLIC port". The all discussion here is around:

1.  agreement on having the "source PUBLIC port"
2.  discussion on also having the "source PRIVATE port"

I'm still not sure, even after this: https://github.com/camaraproject/EdgeCloud/pull/249#discussion_r1637571944 if you are asking for the PUBLIC or PRIVATE or both :-). Sorry it is totally my fault not being an expert in this field.

Reply:
@FabrizioMoggio the UPF looks at source PRIVATE Port since the translation from PRIVATE to PUBLIC happens after UPF processing (in Uplink direction which is what we are concerned with for Traffic Influence).

Regarding what ports to include in the request, my comment is that both are required:
PUBLIC port is required to identify the device.  It can be part of the Device identifier attribute. 
PRIVATE port is required to identify the traffic. This can be part of sourceTrafficFilter attribute.

Reply:
dear @babunkj thank you so much, that is correct, in a traditional scenario, you are totally right. Now I got you.

My confusion is due to the discussion we are having in FWA (or tethering), that is what @gunjald was referring to, according to my understanding. In FWA we have a device with a "Private-1 IP/Port" connected to a FWA device. The FWA device translate it to "Private-2 IP/Port". This is what the UPF gets. Than we have CGAT and so "Public IP/Port"

Private-1 (in the LAN)
Private-2 (Core Network)
Public (Internet)

I was focusing on Private-1 forgetting about Private-2 :-).

So we currently we have:

Device: with Private-1-IP and and Public-IP and Public-Port

Where Private-1=Private-2 for a smartphone. In this scenario, the common one, we need to introduce, I agree with you, Private-1-Port. Because this is the Port the UPF uses.

This is not enough for FWA or tethering. In this scenario Private-1 is different from Private-2 and we also need Private-2-IP and Private-2-Port. In this scenario indeed the UPF works with Private-2-IP and Private-2-Port.

About Private-1-IP: it is provided by the client to the API Consumer that puts it into Device
About Public-IP: it is discovered by the API Consumer that puts it into Device
About Private-2-IP: it can be discovered by the Network, knowing Public-IP

What about Private-2-Port? how can we get it?

Reply:
Right @FabrizioMoggio. That is an interesting scenario for FWA and mobile devices which act as hotspot. 
I think the Private-2-IP and Private-2-Port you referred to may be derived from the Public IP and Public Port (Reverse mapping in NAT). However, these values are not visible to the Application server.